lib/apiservers/engine/backends/container/viccontainer.go

```diff
@@ -65,3 +71,21 @@ func (v *VicContainer) List() []string {
 	v.m.RUnlock()
 	return IDs
 }
+
+// Tries to lock the container.  Timeout argument defines how long the lock
+// attempt will be tried.  Returns true if locked, false if timed out.
+func (v *VicContainer) TryLock(timeout time.Duration) bool {
+	timeChan := time.After(timeout)
+	select {
+	case <-timeChan:
+		return false
+	case <-v.lockChan:
+		return true
+
+	}
+}
+
+// Unlocks the container
+func (v *VicContainer) Unlock() {
+	v.lockChan <- true
```

Comment:
What happens if you accidentally call Unlock twice?

Reply:
That would be a problem.  This is a super simple lock with timeout.  Let me rethink this.

Reply:
You could use select to read from the channel and panic if default works.
My point is that unexpected lock is hard to debug, if it panics here, it would be a lot easier.
Standard mutex doesn't allow you to unlock things twice, it panics.